packages/sdk-core/tasks/build-types.sh

```diff
@@ -0,0 +1,22 @@
+#!/usr/bin/env bash
+
+# make sure that if any step fails, the script fails
+set -xe
+
+if [ -d "./src/typechain-types" ]; then
+  rm -rf ./src/typechain-types
+fi
+
+# if the typechain files do not exist, we build
+# hardhat so that it does exist
+if [ ! -d "../ethereum-contracts/typechain-types" ]; then
+  cd ../ethereum-contracts
+  yarn build:contracts:hardhat
```

Comment:
leaking abstraction: a nicer error message when typechain-types not available suffices.